src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -472,19 +530,27 @@ public function dumpFile($filename, $content, $mode = 0666)
             throw new IOException(sprintf('Unable to write to the "%s" directory.', $dir), 0, null, $dir);
         }
 
-        $tmpFile = tempnam($dir, basename($filename));
+        $tmpFile = $this->tempNam($dir, basename($filename));
```

Comment:
Instead of creating a custom tempNam implementation, wouldn't it be better to bypass the temporary file creation and directly write to $filename, WHEN a stream wrapper is used?